### ISSUE TITLE
Font rendering on custom border characters

### ISSUE THREAD

**Issue by seblj** [ghostty-org/ghostty#3415]
Hi :)

# Problem

I have created a custom font of the rounded border characters to make them "filled". I really like this look, as it makes popups more "modern" in my eyes.

This works great in kitty, but doesn't work as expected in ghostty.

## Kitty:
<img width="219" alt="Image" src="https://github.com/user-attachments/assets/7ddcd308-f1ad-4427-a947-509edf6b2ab2" />

## Ghostty:
<img width="215" alt="Image" src="https://github.com/user-attachments/assets/51022aab-29ca-487a-8f79-b959923b2bd9" />

# General info

I am using macOS 15.1.1 (24B91) and this is the output of `ghostty --version`:
```
Ghostty 1.0.0

Version
  - version: 1.0.0
  - channel: stable
Build Config
  - Zig version: 0.13.0
  - build mode : builtin.OptimizeMode.ReleaseFast
  - app runtime: apprt.Runtime.none
  - font engine: font.main.Backend.coretext
  - renderer   : renderer.Metal
  - libxev     : main.Backend.kqueue
```

# Root cause

Debugging this, I think I found out why this happens, but I am not sure about how to potentially resolve this. Here: https://github.com/ghostty-org/ghostty/blob/a8e5eef11cc67f87f445626f9ca2993373774bf8/src/font/sprite/Box.zig#L251 there seems to be special handling on certain unicodes and how to draw them, and since I have created a custom font and just used some random unicode points (U+e4c3-U+e4d4) they are not handled specially, and therefore not rendered as expected...

I noticed this when trying to copy the rounded border characters to the unicode value I am using, and noticed that they are rendered differently. I also tried to copy my unicode and use some of the characters that are properly handled, and they seem to be rendered as expected then.

However, I don't want to override some very normal border unicode characters.

This is how my border characters look:
 <img width="349" alt="Image" src="https://github.com/user-attachments/assets/651e8c0c-9e8e-488f-8ea9-994070fd6f45" />

and you can find the font here: https://github.com/seblj/dotfiles/blob/master/fonts/corners.ttf

**Comment by qwerasd205** [comment]
https://ghostty.org/docs/config/reference#font-codepoint-map

**Comment by seblj** [comment]
No, that is not the issue. The issue is that the border characters are not rendered specially like the other "normal" border characters.

The font is correctly used

**Comment by qwerasd205** [comment]
Is the issue that they're the wrong size? Perhaps you would want to adjust the cell size down (there's config options for this) so they fit exactly from your font?

**Comment by seblj** [comment]
Yeah maybe. But the same issue is if you for example copy the border characters from U+256d-U2570 and put them in for example U+e400-U+e403, then they will get rendered incorrectly because the switch never happens

**Comment by qwerasd205** [comment]
Ghostty only does custom rendering for known codepoints which need to fill the cell in some way. In the future there may be a way to configure certain codepoints to be automatically stretched to the cell width/height, but that is not currently possible.
